src/clojure/clj_ldap/client.clj

```diff
@@ -43,6 +43,15 @@
 (defn encode [attr]
   (.getValue attr))
 
+(defn ssl-protocol-mapping
+  "Converts user protocol settings into valid string constants"
```

Comment:
It would be good to create a unit test for this that demonstrates the translation, and demonstrates what happens if a user specifies a protocol that is unmapped.